Release failed due to an issue with the ARM build

/kind bug

**What steps did you take and what happened:**

When doing the v2.4.0 release we ran into an issue with the arm build of the controller-manager:

```
179.9 # github.com/openshift/rosa/pkg/ocm
179.9 /go/pkg/mod/github.com/openshift/rosa@v1.2.35-rc1.0.20240221134836-0beb882e5836/pkg/ocm/helpers.go:969:21: math.MaxInt64 (untyped int constant 9223372036854775807) overflows int
```

@damdo investigated and found this reference: https://github.com/golang/go/issues/23086 and this might need to be changed as a result: https://github.com/openshift/rosa/blob/ded22540d1fe227ea0a13197c7a8136099be890e/pkg/ocm/helpers.go#L969C21-L969C34

This highlights that there is a gap in our pre-submit jobs that cover all the architectures we support.

**What did you expect to happen:**

Expect the release process to work.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

--- a/go.mod
+++ b/go.mod
@@ -29,8 +29,8 @@ require (
 	github.com/onsi/ginkgo/v2 v2.13.1
 	github.com/onsi/gomega v1.30.0
 	github.com/openshift-online/ocm-common v0.0.0-20240129111424-ff8c6c11d909
-	github.com/openshift-online/ocm-sdk-go v0.1.403
-	github.com/openshift/rosa v1.2.35-rc1.0.20240221134836-0beb882e5836
+	github.com/openshift-online/ocm-sdk-go v0.1.404
+	github.com/openshift/rosa v1.2.35-rc1.0.20240229115423-42874686e22d
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.18.0
 	github.com/sergi/go-diff v1.3.1

